src/librustc/ty/maps/plumbing.rs

```diff
@@ -912,7 +948,8 @@ pub fn force_from_dep_node<'a, 'gcx, 'lcx>(tcx: TyCtxt<'a, 'gcx, 'lcx>,
                         profq_query_msg!(::ty::maps::queries::$query::NAME, tcx, $key),
                     )
                 );
-
+                // Forcing doesn't add a read edge, but executing the query may add read edges.
+                // Could this add a `read edge too?
```

Comment:
We need control over whether a read edge is added or not. In this case (`force_from_dep_node()`) the appropriate edges are added during `try_mark_green()` where there is no source `DepNode` for the edge because creating that node might later fail (if a subsequent input turns up red).

Reply:
This comment was not supposed to be on this PR.